### ISSUE TITLE
arm64: broken test if compiling with clang -Os

### ISSUE THREAD

**Issue by cosmos72** [cosmos72/m4th#2]
since commit ce8f11d4ccc74ccec854923699a65c33ec5c3870 which renamed m4th -> m64th
compiling on arm64 with `make CC='clang -Os'` causes test failures.

the simplest failed test is `"pq" type`:
```console
  I/O   test failed: "pq" type
    expected  data  stack <0>
      actual  data  stack <7> $40f920 2 $4a2860 $4a28a0 $4a2898 1 $4a0a2a
... expected return stack <0>
      actual return stack <7> $40b454 $401a08 $7fdd3784b2 $4030a4 $401b84 $43e91e $4a2860
... expected  in   buffer <0> []
      actual  in   buffer <0> []
... expected  out  buffer <3> [opq]
      actual  out  buffer <1> [o]
```

initial debugging shows that `(vm)` calls `(bye)` prematurely, instead of the correct token

**Comment by cosmos72** [comment]
using new branch [issue_2_arm64_clang_Os](https://github.com/cosmos72/m64th/tree/issue_2_arm64_clang_Os) to incrementally replay the changes introduced in bulk
by commit ce8f11d4ccc74ccec854923699a65c33ec5c3870 in order to find which one caused the issue

**Comment by cosmos72** [comment]
fixed in commit 2aa741ce96a5f6e434259f30e9a7686656838d5a with a workaround that hints at a clang compiler bug on arm64
